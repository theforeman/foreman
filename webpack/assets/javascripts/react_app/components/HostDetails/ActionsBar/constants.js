export const SUPPORTED_ERRORS = {
  host: __('Host'),
  templates: __('Templates'),
  proxies: __('Proxies'),
};
export const API_OPTIONS = { key: 'BUILD_REVIEW' };
