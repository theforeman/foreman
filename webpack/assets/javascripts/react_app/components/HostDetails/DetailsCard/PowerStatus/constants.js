import { translate as __ } from '../../../../common/I18n';

export const POWER_REQURST_KEY = 'HOST_TOGGLE_POWER';
export const POWER_REQUEST_OPTIONS = { key: POWER_REQURST_KEY };
export const BASE_POWER_STATES = { off: __('Off'), on: __('On') };
export const BMC_POWER_STATES = { soft: __('Reboot'), cycle: __('Reset') };
export const SUPPORTED_POWER_STATES = {
  ...BASE_POWER_STATES,
  ...BMC_POWER_STATES,
};
