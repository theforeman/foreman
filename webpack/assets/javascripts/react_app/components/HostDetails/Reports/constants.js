export const REPORT_API_OPTIONS = { key: 'HOST_DETAILS_RECENT_REPORTS' };
