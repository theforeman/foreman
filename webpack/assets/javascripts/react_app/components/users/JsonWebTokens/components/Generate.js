import React from 'react';
import PropTypes from 'prop-types';
import * as Yup from 'yup';

import { Button } from 'patternfly-react';

import ForemanModal from '../../../ForemanModal';
import ForemanForm from '../../../common/forms/ForemanForm';
import DateTime from '../../../common/forms/DateTime/DateTime';
import { GENERATE_JWT_MODAL_ID } from '../Constants';

import { translate as __ } from '../../../../common/I18n';

const Generate = ({ handleSubmit, modalActions }) => {
  const validationSchema = Yup.object().shape({
    expiresAt: Yup.date().min(new Date(), __("Can't be in the past")),
  });

  return (
    <>
      <Button
        bsStyle="success"
        className="btn-lg"
        onClick={modalActions.setModalOpen}
      >
        {__('Generate token')}
      </Button>

      <ForemanModal
        id={GENERATE_JWT_MODAL_ID}
        title={__('Generate JSON web token')}
      >
        <ForemanModal.Header />
        <ForemanForm
          onSubmit={handleSubmit}
          initialValues={{}}
          validationSchema={validationSchema}
          onCancel={modalActions.setModalClosed}
        >
          <DateTime
            id="jwt_expires_at"
            label={__('Expires at')}
            isRequired={false}
            inputClassName="col-md-5"
            placement="bottom"
            inputProps={{
              name: 'expiresAt',
              placeholder: __('Never'),
              autoComplete: 'off',
            }}
            value={null}
          />
        </ForemanForm>
      </ForemanModal>
    </>
  );
};

Generate.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  modalActions: PropTypes.object.isRequired,
};

export default Generate;
