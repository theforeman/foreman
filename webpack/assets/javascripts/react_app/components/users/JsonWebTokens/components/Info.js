import React from 'react';
import { translate as __ } from '../../../../common/I18n';

const Info = () => (
  <>
    {__(
      'JSON web token (JWT) allow you to authenticate API requests without exposing your credentials.'
    )}
    <p>
      <code>
        curl &apos;/api/statuses&apos; -H &apos;Authorization: Bearer
        token-value&apos;
      </code>
    </p>
  </>
);

export default Info;
