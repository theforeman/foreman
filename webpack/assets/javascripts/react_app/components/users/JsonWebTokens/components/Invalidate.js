import React from 'react';
import PropTypes from 'prop-types';

import { Button } from 'patternfly-react';

import { translate as __ } from '../../../../common/I18n';
import ForemanModal from '../../../ForemanModal';
import { INVALIDATE_JWT_MODAL_ID } from '../Constants';

const Invalidate = ({ handleSubmit, modalActions }) => (
  <>
    <Button
      bsStyle="danger"
      className="btn-lg"
      onClick={modalActions.setModalOpen}
    >
      {__('Invalidate tokens')}
    </Button>

    <ForemanModal
      id={INVALIDATE_JWT_MODAL_ID}
      title={__('Invalidate JSON web tokens')}
    >
      <ForemanModal.Header />
      {__('Are you sure you want to invalidate all JSON web tokens?')}
      <ForemanModal.Footer>
        <Button bsStyle="primary" onClick={handleSubmit}>
          {__('Invalidate')}
        </Button>
        &nbsp;
        <Button bsStyle="default" onClick={modalActions.setModalClosed}>
          {__('Cancel')}
        </Button>
      </ForemanModal.Footer>
    </ForemanModal>
  </>
);

Invalidate.propTypes = {
  modalActions: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Invalidate;
