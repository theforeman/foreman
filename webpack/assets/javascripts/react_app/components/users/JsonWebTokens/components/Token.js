import React from 'react';
import PropTypes from 'prop-types';

import ClipboardCopy from '../../../common/ClipboardCopy';
import SkeletonLoader from '../../../common/SkeletonLoader';
import { translate as __ } from '../../../../common/I18n';

const Token = ({ status, token }) => (
  <SkeletonLoader status={status} emptyState="">
    <div className="text-center">
      <pre>{token}</pre>
      <ClipboardCopy
        text={token}
        buttonText={__('Copy token to clipboard')}
        textareaProps={{
          hidden: true,
        }}
        buttonProps={{ className: 'center-block' }}
      />
    </div>
  </SkeletonLoader>
);

Token.propTypes = {
  status: PropTypes.string,
  token: PropTypes.string,
};

Token.defaultProps = {
  status: '',
  token: '',
};

export default Token;
