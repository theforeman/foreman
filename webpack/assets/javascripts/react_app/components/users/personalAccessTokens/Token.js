import React from 'react';
import TimeAgo from '../../common/TimeAgo';
import Button from '../../common/forms/Button';

/* eslint-disable camelcase */
export default ({
  id,
  name,
  created_at,
  expires_at,
  last_used_at,
  user_id,
  revocable,
  revokeToken
}) => (
  <tr>
    <td>{name}</td>
    <td>
      <TimeAgo date={created_at} />
    </td>
    <td>
      <TimeAgo date={expires_at} />
    </td>
    <td>
      <TimeAgo date={last_used_at} />
    </td>
    <td>
      {revocable && (
        <Button onClick={revokeToken} className="btn-sm btn-default">{__('Revoke')}</Button>
      )}
    </td>
  </tr>
);
