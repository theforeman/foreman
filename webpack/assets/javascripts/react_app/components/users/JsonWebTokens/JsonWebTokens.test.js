import { STATUS } from '../../../constants';
import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

import Generate from './components/Generate';
import Info from './components/Info';
import Invalidate from './components/Invalidate';
import Token from './components/Token';

export const generateProps = {
  handleSubmit: () => {},
  modalActions: {
    setModalOpen: () => {},
    setModalClosed: () => {},
  },
  expiresAt: new Date(),
  setExpiresAt: () => {},
}
export const invalidateProps = {
  handleSubmit: () => {},
  modalActions: {
    setModalOpen: () => {},
    setModalClosed: () => {},
  },
}

export const tokenProps = {
  status: STATUS.RESOLVED,
  token: 'token-value'
}

describe('JsonWebTokens', () => {
  describe('rendering', () => {
    testComponentSnapshotsWithFixtures(Generate, {'renders Generate': generateProps});
    testComponentSnapshotsWithFixtures(Info, {'renders Info': {}});
    testComponentSnapshotsWithFixtures(Invalidate, {'renders Invalidate': invalidateProps});
    testComponentSnapshotsWithFixtures(Token, {'renders Token': tokenProps});
  });
});
