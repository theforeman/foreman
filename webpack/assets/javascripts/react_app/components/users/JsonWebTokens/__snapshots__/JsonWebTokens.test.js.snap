// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonWebTokens rendering renders Generate 1`] = `
<Fragment>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="btn-lg"
    disabled={false}
    onClick={[Function]}
  >
    Generate token
  </Button>
  <ConnectedForemanModal
    id="GENERATE_JWT_MODAL_ID"
    title="Generate JSON web token"
  >
    <ForemanModalHeader />
    <ForemanForm
      enableReinitialize={false}
      initialValues={Object {}}
      onCancel={[Function]}
      onSubmit={[Function]}
      validationSchema={
        ObjectSchema {
          "_blacklist": RefSet {
            "list": Array [],
            "refs": Array [],
          },
          "_conditions": Array [],
          "_defaultDefault": [Function],
          "_deps": Array [],
          "_excludedEdges": Array [],
          "_exclusive": Object {},
          "_mutate": undefined,
          "_nodes": Array [
            "expiresAt",
          ],
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "object",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Array [],
            "refs": Array [],
          },
          "fields": Object {
            "expiresAt": DateSchema {
              "_blacklist": RefSet {
                "list": Array [],
                "refs": Array [],
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "min": true,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "date",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Array [],
                "refs": Array [],
              },
              "tests": Array [
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "date",
            },
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
          "type": "object",
        }
      }
    >
      <DateTime
        id="jwt_expires_at"
        inputClassName="col-md-5"
        inputProps={
          Object {
            "autoComplete": "off",
            "name": "expiresAt",
            "placeholder": "Never",
          }
        }
        isRequired={false}
        label="Expires at"
        locale={null}
        placement="bottom"
        value={null}
      />
    </ForemanForm>
  </ConnectedForemanModal>
</Fragment>
`;

exports[`JsonWebTokens rendering renders Info 1`] = `
<Fragment>
  JSON web token (JWT) allow you to authenticate API requests without exposing your credentials.
  <p>
    <code>
      curl '/api/statuses' -H 'Authorization: Bearer token-value'
    </code>
  </p>
</Fragment>
`;

exports[`JsonWebTokens rendering renders Invalidate 1`] = `
<Fragment>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="btn-lg"
    disabled={false}
    onClick={[Function]}
  >
    Invalidate tokens
  </Button>
  <ConnectedForemanModal
    id="INVALIDATE_JWT_MODAL_ID"
    title="Invalidate JSON web tokens"
  >
    <ForemanModalHeader />
    Are you sure you want to invalidate all JSON web tokens?
    <ForemanModalFooter>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        disabled={false}
        onClick={[Function]}
      >
        Invalidate
      </Button>
      Â 
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}
      >
        Cancel
      </Button>
    </ForemanModalFooter>
  </ConnectedForemanModal>
</Fragment>
`;

exports[`JsonWebTokens rendering renders Token 1`] = `
<SkeletonLoader
  emptyState=""
  skeletonProps={Object {}}
  status="RESOLVED"
>
  <div
    className="text-center"
  >
    <pre>
      token-value
    </pre>
    <ClipboardCopy
      buttonProps={
        Object {
          "className": "center-block",
        }
      }
      buttonText="Copy token to clipboard"
      successMessage="Copied!"
      text="token-value"
      textareaProps={
        Object {
          "hidden": true,
        }
      }
    />
  </div>
</SkeletonLoader>
`;
